### ISSUE TITLE
[Bug-Candidate]: False positive for unprotected-upgrade detector when initializers are disabled

### ISSUE THREAD

**Issue by bmarcot** [crytic/slither#1447]
### Describe the issue:

It seems `unprotected-upgrade` can be detected on contracts that have disabled initializers in their constructor. See the following example:

```
//SPDX-License-Identifier: Unlicense
pragma solidity ^0.8.0;

import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";

interface IUniswapV2 {
    function swapExactTokensForTokens(
        uint256 amount,
        address token0,
        address token1,
        address receiver
    ) external returns (uint256[] memory);
}

contract SlitherNotUnprotectedUpgrade is Initializable {
    address private immutable _router;

    constructor(address router) {
        _disableInitializers();
        _router = router;
    }

    function initialize() public initializer {
        // solhint-disable-prev-line no-empty-block
    }

    function swapExactTokensForTokens(
        uint256 amount,
        address token0,
        address token1,
        address receiver
    ) private returns (uint256[] memory) {
        // solhint-disable-next-line avoid-low-level-calls
        (bool success, bytes memory returndata) = _router.delegatecall(
            abi.encodeCall(IUniswapV2.swapExactTokensForTokens, (amount, token0, token1, receiver))
        );
        if (!success) {
            // solhint-disable-next-line no-inline-assembly
            assembly {
                revert(add(returndata, 32), mload(returndata))
            }
        }
        return abi.decode(returndata, (uint256[]));
    }
}
```

And detection by `slither`:

```
ethsec@7d5131fcbad1:/share$ slither .
SlitherNotUnprotectedUpgrade.swapExactTokensForTokens(uint256,address,address,address) (contracts/SlitherNotUnprotectedUpgrade.sol#27-44) uses delegatecall to a input-controlled function id
	- (success,returndata) = _router.delegatecall(abi.encodeCall(IUniswapV2.swapExactTokensForTokens,(amount,token0,token1,receiver))) (contracts/SlitherNotUnprotectedUpgrade.sol#34-36)
Reference: https://github.com/crytic/slither/wiki/Detector-Documentation#controlled-delegatecall
```

It looks like OpenZeppelin introduced `_disableInitializers` specifically for EIP-1167 proxied contracts:

> It is recommended to use this to lock implementation contracts that are designed to be called through proxies.

Ref. https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/proxy/utils/Initializable.sol#L136-L143

I unit tested and I cannot call initilizer on implementation after calling `_disableInitializer`, so I believe this detection is a false positive.  Also I think the problem lies in this code https://github.com/crytic/slither/blob/d8e526e53ff690bd24c260042117efab5ce9c271/slither/detectors/statements/unprotected_upgradeable.py#L25-L39

I can fix if we confirm the issue.

### Code example to reproduce the issue:

//SPDX-License-Identifier: Unlicense
pragma solidity ^0.8.0;

import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";

interface IUniswapV2 {
    function swapExactTokensForTokens(
        uint256 amount,
        address token0,
        address token1,
        address receiver
    ) external returns (uint256[] memory);
}

contract SlitherNotUnprotectedUpgrade is Initializable {
    address private immutable _router;

    constructor(address router) {
        _disableInitializers();
        _router = router;
    }

    function initialize() public initializer {
        // solhint-disable-prev-line no-empty-block
    }

    function swapExactTokensForTokens(
        uint256 amount,
        address token0,
        address token1,
        address receiver
    ) private returns (uint256[] memory) {
        // solhint-disable-next-line avoid-low-level-calls
        (bool success, bytes memory returndata) = _router.delegatecall(
            abi.encodeCall(IUniswapV2.swapExactTokensForTokens, (amount, token0, token1, receiver))
        );
        if (!success) {
            // solhint-disable-next-line no-inline-assembly
            assembly {
                revert(add(returndata, 32), mload(returndata))
            }
        }
        return abi.decode(returndata, (uint256[]));
    }
}

### Version:

0.8.3

**Comment by 0xalpharush** [comment]
This should be fixed if you install slither-analyzer==0.9.0 which includes https://github.com/crytic/slither/pull/1344.

**Comment by bmarcot** [comment]
Fixed in 0.9.0
